common/src/main/java/com/wynntils/wynn/model/quests/QuestManager.java

```diff
@@ -0,0 +1,190 @@
+        // when entering a world
+        if (e.getNewState() == WorldStateManager.State.WORLD) {
+            CONTAINER_QUERIES.queryQuestBook();
+        }
+    }
+
+    public static void rescanQuestBook(boolean includeQuests, boolean includeMiniQuests) {
+        WynntilsMod.info("Requesting rescan of Quest Book");
+        if (includeQuests) {
+            CONTAINER_QUERIES.queryQuestBook();
+        }
+        if (includeMiniQuests) {
+            CONTAINER_QUERIES.queryMiniQuests();
+        }
+    }
+
+    public static void rescanDialogueHistory() {
+        DIALOGUE_HISTORY_QUERIES.scanDialogueHistory();
+    }
+
+    public static List<QuestInfo> getQuests() {
+        return quests;
+    }
+
+    public static List<QuestInfo> getMiniQuests() {
+        return miniQuests;
+    }
+
+    public static List<QuestInfo> getQuestsSorted(QuestSortOrder sortOrder) {
+        return sortQuestInfoList(sortOrder, quests);
+    }
+
+    public static List<QuestInfo> getMiniQuestsSorted(QuestSortOrder sortOrder) {
+        return sortQuestInfoList(sortOrder, miniQuests);
+    }
+
+    private static List<QuestInfo> sortQuestInfoList(QuestSortOrder sortOrder, List<QuestInfo> questList) {
+        // All quests are always sorted by status (available then unavailable), and then
+        // the given sort order, and finally a third way if the given sort order is equal.
+        return switch (sortOrder) {
+            case LEVEL -> questList.stream()
+                    .sorted(Comparator.comparing(QuestInfo::getStatus)
+                            .thenComparing(QuestInfo::getLevel)
+                            .thenComparing(QuestInfo::getName))
+                    .toList();
+            case DISTANCE -> questList.stream()
+                    .sorted(Comparator.comparing(QuestInfo::getStatus)
+                            .thenComparing(new LocationComparator())
+                            .thenComparing(QuestInfo::getName))
+                    .toList();
+            case ALPHABETIC -> questList.stream()
+                    .sorted(Comparator.comparing(QuestInfo::getStatus)
+                            .thenComparing(QuestInfo::getName)
+                            .thenComparing(QuestInfo::getLevel))
+                    .toList();
+        };
+    }
+
+    public static Optional<QuestInfo> getQuestFromName(String name) {
+        return quests.stream().filter(quest -> quest.getName().equals(name)).findFirst();
+    }
+
+    public static List<List<String>> getDialogueHistory() {
+        return dialogueHistory;
+    }
+
+    public static void toggleTracking(QuestInfo questInfo) {
+        CONTAINER_QUERIES.toggleTracking(questInfo);
+    }
+
+    public static void openQuestOnWiki(QuestInfo questInfo) {
+        final String baseUrl = "https://wynncraft.fandom.com/wiki/";
+
+        // TODO handle mini quest
+        String name = questInfo.getName();
+        String wikiQuestPageNameQuery = WebManager.getApiUrl("WikiQuestQuery");
```

Comment:
check if apiurls exists before using

Reply:
Sigh, the apiurls overhead is just bad. :(

Fixed it. Also noticed that the url pointed to the old domain...